sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -0,0 +1,347 @@
+
+```ts
+/**
+ * The default credential first checks environment variables for configuration.
+ * If environment configuration is incomplete, it will try managed identity.
+ */
+function withDefaultAzureCredential() {
+  const credential = new DefaultAzureCredential();
+  const client = new SecretClient(`https://key-vault-name.vault.azure.net`, credential);
+}
+```
+
+### Authenticating a user assigned managed identity with `DefaultAzureCredential`
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `DefaultAzureCredential`, deployed to an Azure resource with a user assigned managed identity configured.
+
+For more information about how to configure a user assigned managed identity for an Azure resource please refer to [What are managed identities for Azure resources][azure_managed_identities].
+
+```ts
+/**
+ * The default credential will use the user assigned managed identity with the specified client ID.
+ */
+function withDefaultAzureCredential() {
+  // Alternatively, you may set the environment variable AZURE_CLIENT_ID="<MANAGED_IDENTITY_CLIENT_ID>" and omit the `managedIdentityClientId`
+  // option when using `DefaultAzureCredential` - the two approaches are equivalent.
+  const credential = new DefaultAzureCredential({
+    managedIdentityClientId: "<MANAGED_IDENTITY_CLIENT_ID>"
+  });
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a user account interactively in the browser
+
+For clients that have a default browser available and for client-side applications running in the browser, the `InteractiveBrowserCredential` provides the simplest user authentication experience. In the sample below an application authenticates a `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] using the `InteractiveBrowserCredential`.
+
+> For client side applications running in the browser, the `InteractiveBrowserCredential` is the only credential type that is supported.
+
+```ts
+function withInteractiveBrowserCredential() {
+  const credential = new InteractiveBrowserCredential({
+    tenantId: "<YOUR_TENANT_ID>",
+    clientId: "<YOUR_CLIENT_ID>"
+  });
+
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+See more about how to configure an AAD application for interactive browser authentication in [Single-page application: App registration](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration)
```

Comment:
Doesnt this apply only for client side applications? Service side apps don't need the app registration when using InteractiveBrowserCredential.

Reply:
I actually had no idea, but yes this is true according to @sadasant . That also makes me more inclined to agree with this suggestion: https://github.com/Azure/azure-sdk-for-js/pull/14646/files/99edb217ed79552fc329a9aa981f72fdeafa88c9#r606346062 as it now gives us a place to add this link / info only in a browser context. Thanks!

Reply:
I hope you don't mind me chiming in in this chimney.

On Node.js, **an AAD application is required for these credentials**:

- AuthorizationCodeCredential 
- ClientSecretCredential.
- ClientCertificateCredential.
- DefaultAzureCredential (if used locally with environment variables set).
- EnvironmentCredential.

On Node.js, **one credential needs an entirely new user to authenticate**:
- UsernamePasswordCredential.

On browsers, **the only credential supported does need an AAD application**:
- InteractiveBrowserCredential.

On Node.js, the **credentials that don't need an AAD application** are:
- ManagedIdentityCredential (which is not designed to work locally).
- AzureCliCredential (which needs `az login`).
- DefaultAzureCredential (which can be used without an AAD app after running `az login`).
- InteractiveBrowserCredential
- DeviceCodeCredential

Sorry if some of this was obvious 😵, I wanted to be thorough to ensure I wasn't skipping something while writing this.